### PR TITLE
add initial tests for inletsoperator_app

### PR DESCRIPTION
**Pull request by cpanato**
## Description
Add initial tests per follow up comment https://github.com/alexellis/arkade/pull/367#pullrequestreview-637506447

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

- Fetch this PR and then build it locally with `make build`.
- Have an inlet license
- Create the token for the provider
- install the inletsoperator with the following command

```
./arkade install inlets-operator --license-file inlet-license --token-file my-token --provider hetzner --region hel1  
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Review by alexellis — CHANGES_REQUESTED** [review]
My comments were never sent, sorry about that. There were left a few minutes after your PR, and now I can see why you didn't respond.

**Comment by alexellis** [comment]
> How Has This Been Tested?

Only part that's missing now.

Please do a quick end to end test with a couple of provider options. You can just install into Kind/K3d without checking that the tunnel is created and a fake access token.

```
Name:	Tester
Email:	contact@openfaas.com
Expires in: 2 days, on: 14 May 2021
Products: [inlets-pro]

Note: use text under the line
---
eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiVGVzdGVyIiwiZW1haWxfYWRkcmVzcyI6ImNvbnRhY3RAb3BlbmZhYXMuY29tIiwicHJvZHVjdHMiOlsiaW5sZXRzLXBybyJdLCJhdWQiOiJqd3QtbGljZW5zZSIsImV4cCI6MTYyMDk5MTU3OSwianRpIjoiODA4MSIsImlhdCI6MTYyMDgxODc3OSwiaXNzIjoiand0LWxpY2Vuc2UiLCJzdWIiOiJUZXN0ZXIifQ.u3n0lyu7LHIzgpPigw3gk_pY8vgWsdxpAyxOMD3LXZFNl6ISX549AOjyC-iFy79DEYVMuKYzoeTQ7-5pz2eHeQ
```

File: cmd/apps/inletsoperator_app_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+package apps
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func Test_wrongFlag(t *testing.T) {
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app_test.go:13]: We don't need this test, can you remove it please?

File: cmd/apps/inletsoperator_app_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	if err.Error() != errWant {
+		t.Errorf("expected error: %s, got: %s", errWant, err)
+	}
+}
+
+func Test_ProviderHetzner(t *testing.T) {
+	cmdInlet := MakeInstallInletsOperator()
+
+	// missing required flags
+	err := cmdInlet.ParseFlags([]string{"--provider", "hetzner"})
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app_test.go:32]: `cmdInlet` -> `cmdInlets`

File: cmd/apps/inletsoperator_app_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	if err != nil {
+		t.Errorf("expected no error when parsing flags, but got: %s", err.Error())
+	}
+
+	errWant := "invalid region set for provider hetzner. Valid regions are: fsn1,nbg1,hel1"
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app_test.go:37]: can you comma separate these values?

Something like:
```
strings.TrimEnd(string.Join(array, ", "), ", ")
```

**Comment by cpanato** [comment]
> > How Has This Been Tested?
> 
> Only part that's missing now.
> 
> Please do a quick end to end test with a couple of provider options. You can just install into Kind/K3d without checking that the tunnel is created and a fake access token.
> 
> ```
> Name:	Tester
> Email:	contact@openfaas.com
> Expires in: 2 days, on: 14 May 2021
> Products: [inlets-pro]
> 
> Note: use text under the line
> ---
> eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiVGVzdGVyIiwiZW1haWxfYWRkcmVzcyI6ImNvbnRhY3RAb3BlbmZhYXMuY29tIiwicHJvZHVjdHMiOlsiaW5sZXRzLXBybyJdLCJhdWQiOiJqd3QtbGljZW5zZSIsImV4cCI6MTYyMDk5MTU3OSwianRpIjoiODA4MSIsImlhdCI6MTYyMDgxODc3OSwiaXNzIjoiand0LWxpY2Vuc2UiLCJzdWIiOiJUZXN0ZXIifQ.u3n0lyu7LHIzgpPigw3gk_pY8vgWsdxpAyxOMD3LXZFNl6ISX549AOjyC-iFy79DEYVMuKYzoeTQ7-5pz2eHeQ
> ```

i will have time more in the end of week/weekend to test this out

**Comment by cpanato** [comment]
@alexellis tested and works fine with `hetzner` and tried as well with `digitalocean`

File: cmd/apps/inletsoperator_app_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	"os"
+	"reflect"
+	"testing"
+)
+
+func Test_ProviderHetzner(t *testing.T) {
+	cmdInlets := MakeInstallInletsOperator()
+
+	// missing required flags
+	err := cmdInlets.ParseFlags([]string{"--provider", "hetzner"})
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app_test.go:17]: I can't see the comment now, but I'm sure I advised not to test cobra itself, just the validation method we extracted.

I'll merge anyway to get this in.

File: cmd/apps/inletsoperator_app_test.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	errWant := "failed to check the token file invalid-token-file: stat invalid-token-file: no such file or directory"
+	err = validatePreRun("", "invalid-token-file", "")
+	if err == nil {
+		t.Errorf("expected error when the token file does not exist")
+	}
+	if err.Error() != errWant {
+		t.Errorf("expected error: %s, got: %s", errWant, err)
+	}
+
+	errWant = "failed to check the secret file invalid-secret-file: stat invalid-secret-file: no such file or directory"
```
Comment (alexellis) [inline cmd/apps/inletsoperator_app_test.go:85]: Each of these needs to be a different test fixture for maintainability or a separate case in a test table, if it lends itself to that.

File: cmd/apps/inletsoperator_app_test.go (hunk 4 above)
Comment (cpanato) [reply cmd/apps/inletsoperator_app_test.go:17]: will remove the test

**Review by alexellis — APPROVED** [review]
The test table looks more maintainable. Thank you for the work on this.

**Pull request merged**